holoviews/streams.py

```diff
@@ -391,6 +392,152 @@ def hashkey(self):
         return {'hash': uuid.uuid4().hex}
 
 
+class Buffer(Pipe):
+    """
+    Buffer allows streaming and accumulating incoming chunks of rows
+    from tabular datasets. The data may be in the form of a pandas
+    DataFrame, 2D arrays of rows and columns or dictionaries of column
+    arrays. Buffer will accumulate the last N rows, where N is defined
+    by the specified ``backlog``. The accumulated data is then made
+    available via the ``data`` parameter.
+
+    A Buffer may also be instantiated with a streamz.StreamingDataFrame
+    or a streamz.StreamingSeries, it will automatically subscribe to
+    events emitted by a streamz object.
+
+    When streaming a DataFrame will use the DataFrame index by
+    default, this may be disabled by setting index=False.
```

Comment:
Use the DataFrame index how? As in simply retain it or something more sophisticated?